httpie/uploads.py

```diff
@@ -64,13 +67,53 @@ def wrapped(*args, **kwargs):
     return wrapped
 
 
+def is_stdin(file: IO) -> bool:
+    try:
+        file_no = file.fileno()
+    except Exception:
+        return False
+    else:
+        return file_no == sys.stdin.fileno()
+
+
+READ_THRESHOLD = float(os.getenv("HTTPIE_STDIN_READ_THRESHOLD", 10.0))
+
+
+def trigger_stdin_warning_thread(env: Environment, file: IO) -> None:
+    # Windows unfortunately does not support select() operation
+    # on regular files, like stdin in our use case.
+    # https://docs.python.org/3/library/select.html#select.select
+    if os.name == 'nt':
+        return None
```

Comment:
We have `httpie.compat.is_windows()`. If it’s more nuanced, I’d still probably add a named check.